src/vs/workbench/contrib/terminal/browser/xterm/contextualActionAddon.ts

```diff
@@ -84,6 +86,7 @@ export class ContextualActionAddon extends Disposable implements ITerminalAddon,
 			return;
 		}
 		this._register(commandDetection.onCommandFinished(async command => {
+			this._decoration?.dispose();
```

Comment:
I think we actually want to do it on `onCommandFinished` or `onCommandExecuted, whichever comes first`. This doesn't exist right now but we can fire it on the last line of `handleCommandExecuted`.

Reply:
since it'll happen with whichever comes first, safe to listen to both imo

Reply:
Yep, can then do this safely:

```ts
this._decoration?.dispose();
this._decoration = undefined;
```

Reply:
seems sketchy to do that for windows, so will only follow command executed for non-windows

Reply:
Why is that? The executed sequence should still only fire when you press enter and the command starts executing. We don't care about the positioning, that's the part that's flaky on Windows

Reply:
yes you're right - I was recalling this code which is unrelated to command executed

https://github.com/microsoft/vscode/blob/8624f3a4e39cd23214237a3f6e7046bbb83a0bbb/src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts#L358-L377